tests/ut/ir/operators/test_gather_compare.py

```
@@ -0,0 +1,168 @@
+        assert "tile.gather_compare" in str(prog)
+
+    def test_gt_with_offset(self):
+        prog = _build_program(cmp_mode="gt", offset=4)
+        assert "tile.gather_compare" in str(prog)
+
+    def test_int_cmp_mode(self):
+        prog = _build_program(cmp_mode=2)  # lt
+        assert "tile.gather_compare" in str(prog)
+
+    def test_invalid_cmp_mode_string(self):
+        with pytest.raises(Exception):
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)